lib/DDGC/Web/Service/ActivityFeed.pm

```diff
@@ -0,0 +1,86 @@
+package DDGC::Web::Service::ActivityFeed;
+
+# ABSTRACT: Community Activity JSON service
+
+=pod
+
+=head1 NAME
+
+DDGC::Web::Service::ActivityFeed - Log / retrieve Community Activity.
+
+=head1 DESCRIPTION
+
+This service logs any activity you care to generate and allows subscription
+to these activities via a key of your choosing.
+
+=head1 MOUNT POINT
+
+This service is expected to be mounted on '/activityfeed.json'
+
+=head1 REQUEST HANDLERS
+
+=cut
+
+use DDGC::Base::Web::Service;
+use Try::Tiny;
+use List::MoreUtils qw/ uniq /;
+
+sub pagesize { 40 }
+
+sub activity_page {
+    my ( $params ) = @_;
+    $params->{page} //= 1;
+    $params->{pagesize} //= pagesize;
+    rset('ActivityFeed')->search_rs({
+        ( $params->{filter} )
+            ? ( type => { -like => $params->{filter} =~ s/\*/%/gr } )
+            : (),
+        for_user => { '=' => [ uniq( undef, $params->{user} ) ] }
+    }, {
+        order_by => { -desc => 'me.id' },
+        rows     => $params->{pagesize},
+        page     => $params->{page},
+    });
```

Comment:
Apropos, popped up in the timeline JUST now:

https://twitter.com/OvidPerl/status/624140719968555008 :)